### ISSUE TITLE
SupressWarnings on inner classes issue

### ISSUE THREAD

**Issue by ZacSweers** [google/error-prone#550]
Tried using `SuppressWarnings` to suppress a false positive, but EP seems to just ignore it. Example case:

I have a class A with an inner class B. Error prone warns that B can be static because it (incorrectly, will file a bug with a repro case later) infers it does not access methods in A. When I try to suppress the `ClassCanBeStatic` check on B though, it seems to be ignored.

Basic stencil

```java
class A {
  @SuppressWarnings("ClassCanBeStatic")
  private class B {

  }
}
```

I'm not sure if this is just not supported or a bug.

**Comment by hzsweers** [comment]
Quick update - it does ignore it if I put the suppression on `A` itself

**Comment by cushon** [comment]
I can't reproduce this:

```
$ cat A.java
class A {
  private class B {}
}
$ java -cp error_prone_ant.jar com.google.errorprone.ErrorProneCompiler A.java
A.java:2: warning: [ClassCanBeStatic] Inner class is non-static but does not reference enclosing class
  private class B {}
...
$ cat A.java
class A {
  @SuppressWarnings("ClassCanBeStatic")
  private class B {}
}
$ java -cp error_prone_ant.jar com.google.errorprone.ErrorProneCompiler A.java
```

**Comment by hzsweers** [comment]
I have a repro case in a private pet project, emailed you

**Comment by cushon** [comment]
I remember we talked about this a bit outside the bug I wasn't able to repro, please re-open if you're still seeing this.
